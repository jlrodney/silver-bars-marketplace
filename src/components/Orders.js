import React from 'react';
import Order from './Order';

const Orders = (props) => (
  <div>
    {
      props.orders.map((order) => (
        <Order
          key={order.orderId}
          userId={order.userId}
          quantity={order.quantity}
          price={order.price}
          type={order.type}
        />
      ))
    }
  </div>
);

export default Orders;
