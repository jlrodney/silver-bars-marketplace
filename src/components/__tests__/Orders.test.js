import React from 'react';
import { shallow, mount } from 'enzyme';
import Orders from '../Orders';

const buyOrders = [
  {
    orderId: 'xyz986',
    userId: '1',
    quantity: 22,
    price: 230,
    type: 'BUY'
  },
  {
    orderId: 'xyz987',
    userId: '3',
    quantity: 32,
    price: 2300,
    type: 'BUY'
  }
]

const sellOrders = [
  {
    orderId: 'xyz998',
    userId: '1',
    quantity: 222,
    price: 230,
    type: 'SELL'
  },
  {
    orderId: 'xyz989',
    userId: '2',
    quantity: 22,
    price: 230,
    type: 'SELL'
  }
]
describe('Orders component', () => {
  it('should render Orders with indivual orders', () => {
    const wrapper = shallow(<Orders orders={buyOrders} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Orders with indivual orders', () => {
    const wrapper = shallow(<Orders orders={sellOrders} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Orders with an empty message', () => {
    const wrapper = shallow(<Orders orders={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
