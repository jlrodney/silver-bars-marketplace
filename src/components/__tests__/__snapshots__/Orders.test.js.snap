// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders component should render Orders with an empty message 1`] = `<div />`;

exports[`Orders component should render Orders with indivual orders 1`] = `
<div>
  <Order
    key="xyz986"
    price={230}
    quantity={22}
    type="BUY"
    userId="1"
  />
  <Order
    key="xyz987"
    price={2300}
    quantity={32}
    type="BUY"
    userId="3"
  />
</div>
`;

exports[`Orders component should render Orders with indivual orders 2`] = `
<div>
  <Order
    key="xyz998"
    price={230}
    quantity={222}
    type="SELL"
    userId="1"
  />
  <Order
    key="xyz989"
    price={230}
    quantity={22}
    type="SELL"
    userId="2"
  />
</div>
`;
